### DISCUSSION TITLE
Conciseness in data types

### DISCUSSION THREAD

**Discussion by espenvis** [scalar/scalar#3320] [Ideas]
Hello,

When using Scalar I find that it can often times become very cluttered, because all details of the input data types are expanded (with no option to minimize).

Take for example a data type of an object that has an enum representing countries:
![image](https://github.com/user-attachments/assets/4bc021da-385c-403d-b9c7-e9f4cf82e4fe)
*(the list goes on much further...)*

For enums in particular, showing perhaps a few values, then minimizing if a greater number of values is present, sounds good to me.

With that being said, large objects in the request body alone makes the overview a bit more obnoxious to deal with for a company that has a bunch of large data types thrown around.

Desired behaviour is already present for nested objects:
![image](https://github.com/user-attachments/assets/2b1f306d-3bb6-4c59-9736-58f4f2ba52d7)

This could be applied to top level objects as well. And preferably by default, but I'm happy with a configuration option for that. I would appreciate it if one could also see the actual data type name instead of OpenAPI's generic `object`. I'd also suggest a fixed-height container for the expanded child attributes, where overflow is navigated with a scroller to keep things a bit more compact.

**Comment by antlio** [comment]
hey @espenvis, thanks for that feedback! that a great suggestion, will work on it and keep you updated here

**Reply by espenvis** [reply]
> hey @espenvis, thanks for that feedback! that a great suggestion, will work on it and keep you updated here

That's great! I'm excited to see what you come up with. It's one of the big pet peeves that prevent us from using Scalar over something like Swagger docs. I think for the most part, people are interested in knowing:
 - Query- and path parameters and their accompanying description (if any)
 - The object type of any data types present in the body (i.e. `Customer`) and its accompanying description

Exploring the data type itself is more-so an on-demand need rather than a persistent one. 

You'd probably also want to display the actual data type name over the generic `object` OpenAPI type all other places if possible.

For instance, in the datatypes section, we have an object with a child list of `Attribute` objects. However, we don't know anything about this object besides its properties (it's not listed in the datatype list, nor is it named, you only see `[]object`). If we have two identical or similar structures, it'll be hard to differentiate. (i.e. `CustomerAttribute` vs. `CompanyAttribute`).

So you don't necessarily know all datatype names, but ultimately in a OpenAPI generated client you'd benefit from knowing the object name, as this is what you'll be using in your code, and it'd be easier to find reference documentation if any.

Of course other folks might be able to come with more detailed input and nuanced perspectives here.

**Reply by antlio** [reply]
thanks again these or great feedbacks! will start by this enum QOL and share all of these with the team too!

**Comment by antlio** [comment]
sorry for the time taken regarding this, a first pr #3670 as been open to fix the first issue.

**Reply by espenvis** [reply]
Looks great, perhaps you could also disclose number of additional elements. For example, "Show all 141 values"
